test/unit/object/test_chunked_upload.py

```diff
@@ -134,3 +137,104 @@ def test_start(test_upload_session, mock_box_session):
+        'offset': 6,
+        'size': 1,
+        'sha1': None,
+    }
+    upload_session_mock_object.commit.return_value.json.return_value = {
+        'entries': [
+            {
+                'type': 'file',
+                'id': '12345',
+                'description': 'This is a test description',
+            }
+        ]
+    }
+    mock_iterator.__iter__.return_value = [first_part, fourth_part]
+    upload_session_mock_object.get_parts.return_value = mock_iterator
+    chunked_uploader = ChunkedUploader(upload_session_mock_object, stream, file_size)
+    chunked_uploader.resume()
+    calls = [call(offset=2, part_bytes=b'cd', total_size=7),
+             call(offset=4, part_bytes=b'ef', total_size=7), ]
+    upload_session_mock_object.upload_part_bytes.assert_has_calls(calls, any_order=False)
+
+
+def test_resume_in_process():
+    file_size = 7
+    part_bytes = b'abcdefg'
+    stream = io.BytesIO(part_bytes)
+    upload_session_mock_object = Mock(UploadSession)
+    mock_iterator = MagicMock(LimitOffsetBasedDictCollection)
+    upload_session_mock_object.total_parts = 4
+    upload_session_mock_object.part_size = 2
+    upload_session_mock_object.id = 'F971964745A5CD0C001BZ4E58196BFD'
+    upload_session_mock_object.type = 'upload_session'
+    upload_session_mock_object.num_parts_processed = 0
+    first_part = {
+        'part_id': 'CFEB4BA9',
+        'offset': 0,
+        'size': 2,
+        'sha1': '2iNhTgJGmg18e9G9q1ycR0sZBNw=',
+    }
+    second_part = {
+        'part_id': '4DBB872D',
+        'offset': 2,
+        'size': 2,
+        'sha1': 'A0d4GYoEXB7YC+JxzdApt2h09vw=',
+    }
+    third_part = {
+        'part_id': '6F2D3486',
+        'offset': 4,
+        'size': 2,
+        'sha1': '+CIFFHGVe3u+u4qwiP6b1tFPQmE=',
+    }
+    fourth_part = {
+        'part_id': '4DBC872D',
+        'offset': 6,
+        'size': 1,
+        'sha1': 'VP0XESCfscB4EJI3QTLGbnniJBs=',
+    }
+    uploaded_part_one_mock = Mock()
+    uploaded_part_two_mock = Mock()
+    uploaded_part_four_mock = Mock()
+    uploaded_part_one_mock = {
+        'part': first_part
+    }
+    uploaded_part_two_mock = {
+        'part': second_part
+    }
+    uploaded_part_four_mock = {
+        'part': fourth_part
+    }
+    mock_iterator.__iter__.return_value = [first_part, second_part, third_part]
+    upload_session_mock_object.get_parts.return_value = mock_iterator
+    chunked_uploader = ChunkedUploader(upload_session_mock_object, stream, file_size)
+    upload_session_mock_object.upload_part_bytes.side_effect = [uploaded_part_one_mock, uploaded_part_two_mock,
+                                                                BoxAPIException(502), uploaded_part_four_mock]
+    try:
+        chunked_uploader.start()
+    except BoxAPIException:
+        chunked_uploader.resume()
+    calls = [call(offset=6, part_bytes=b'g', total_size=7)]
+    upload_session_mock_object.upload_part_bytes.assert_has_calls(calls, any_order=False)
```

Comment:
This test should also assert that the upload is committed and have some assertions around the return value of `resume()`